src/libman/Contracts/ConsoleLogger.cs

```diff
@@ -84,5 +88,13 @@ public void Log(string message, LogLevel level)
                 }
             }
         }
+
+        private void ThrowIfInputIsRedirected()
+        {
+            if (Console.IsInputRedirected)
+            {
+                throw new InvalidOperationException();
```

Comment:
nit: maybe have a message in the exception, so that if someone does catch this while debugging, they know what it's from?

Reply:
Fixed. 
Also squashed the changes to a single commit.